db/views/reporting_overdue_patients_v01.sql

```diff
@@ -0,0 +1,89 @@
+  AND appointments.device_created_at AT TIME ZONE 'UTC' AT TIME ZONE (SELECT current_setting('TIMEZONE')) < reporting_patient_states.month_date
+  LEFT JOIN patient_phone_numbers ON patient_phone_numbers.patient_id = reporting_patient_states.patient_id
+  LEFT JOIN lateral (
+    -- Merging BS, BP, Appointments & PD
+    SELECT id AS visit_id, patient_id, recorded_at AS visited_at_after_appointment, 'Blood Sugar' AS visit_type
+    FROM blood_sugars
+    WHERE deleted_at IS NULL AND patient_id = reporting_patient_states.patient_id AND appointments.device_created_at < blood_sugars.recorded_at
+    UNION
+      (SELECT id AS visit_id, patient_id, recorded_at AS visited_at_after_appointment, 'Blood Pressure' AS visit_type
+        FROM blood_pressures
+        WHERE deleted_at IS NULL AND patient_id = reporting_patient_states.patient_id AND appointments.device_created_at < blood_pressures.recorded_at)
+    UNION
+      (SELECT id AS visit_id, patient_id, device_created_at AS visited_at_after_appointment, 'Appointments' AS visit_type
+        FROM appointments AS appointments_visit
+        WHERE deleted_at IS NULL AND patient_id = reporting_patient_states.patient_id AND appointments.device_created_at < appointments_visit.device_created_at)
+    UNION
+      (SELECT id AS visit_id, patient_id, device_created_at AS visited_at_after_appointment, 'Prescription Drugs' AS visit_type
+        FROM prescription_drugs
+        WHERE deleted_at IS NULL AND patient_id = reporting_patient_states.patient_id AND appointments.device_created_at < prescription_drugs.device_created_at)
+  ) visits ON reporting_patient_states.patient_id = visits.patient_id
+  -- Used to determine if the patient was removed from overdue list at the beginning of the month
+  LEFT JOIN call_results AS previous_call_results
+    ON reporting_patient_states.patient_id = previous_call_results.patient_id
+      AND previous_call_results.device_created_at AT TIME ZONE 'UTC' AT TIME ZONE (SELECT current_setting('TIMEZONE')) < reporting_patient_states.month_date
+      AND previous_call_results.device_created_at > appointments.scheduled_date
+  -- Used to determine the outcome of a call and how it affects the rate of patients returing to care
+  FULL JOIN call_results AS next_call_results
+    ON reporting_patient_states.patient_id = next_call_results.patient_id
+      AND next_call_results.device_created_at AT TIME ZONE 'UTC' AT TIME ZONE (SELECT current_setting('TIMEZONE')) >= reporting_patient_states.month_date
+      AND next_call_results.device_created_at AT TIME ZONE 'UTC' AT TIME ZONE (SELECT current_setting('TIMEZONE')) < reporting_patient_states.month_date + interval '1 month'
+WHERE
+  reporting_patient_states.status <> 'dead'
+ORDER BY
+  reporting_patient_states.month_date,
+  reporting_patient_states.patient_id,
+  appointments.device_created_at DESC,
+  visits.visited_at_after_appointment,
+  next_call_results.device_created_at,
+  previous_call_results.device_created_at DESC
```

Comment:
How many months of data do we want to show on the dashboard? Do you think we can restrict this query to only pick, say the last 24 months?

Reply:
On the dashboard, we need only 6 months of data. But we need it to be available for as long as possible for folks to download the line list. I'll check with @jamiecarter7 if that's the case.